languages/ruby/lib/oso/polar/host.rb

```diff
@@ -35,7 +62,7 @@ def initialize_copy(other)
       def get_class(name)
         raise UnregisteredClassError, name unless classes.key? name
 
-        classes[name]
+        classes[name].get
```

Comment:
Could this whole thing just be done in get_class?
We could still check that the class is registered, but then return
```
Object.const_get(name)
```

Wrapping the class in another one adds a lot of this wrapping unwrapping code. Seems like just handling this here would be simpler.

Reply:
That would break `is_a?` though so I guess maybe it's not that simple and we'd have to do some class conversion stuff there too.

Reply:
Yeah the weird thing there is when you want to do a lookup/call on a class like `User.count` or `User.is_admin(user)`, which uses the value returned from `get_instance`.

So if we unwound this abstraction, we'd need to use `const_get` in two places (`get_class` and `get_instance`). I'm not opposed to that, but might also be a bit weird.

Reply:
Also, this reminded me about a particular edge case with anonymous classes (that I just added a test for and fixed), which now makes the abstraction a bit more reasonable imo